joblib/numpy_pickle.py

```diff
@@ -435,7 +435,9 @@ def dump(value, filename, compress=0, protocol=None, cache_size=None):
     else:
         compress_level = compress
 
-    if compress_method == 'lz4' and lz4 is None:
+    # LZ4 compression is only supported and installation checked with
+    # python 3+.
+    if compress_method == 'lz4' and lz4 is None and PY3_OR_LATER:
         raise ValueError(LZ4_NOT_INSTALLED_ERROR)
```

Comment:
This feels weird to have different exception here. Can't we raise `NotImplementedError`?

Reply:
`NotImplementedError` would work for lzma in python2 because it's not part of the standard lib in this version of python.
Here we consider the user wants to dump using lz4 compression, without the package installed, so it's a value error. Python2 is skipped because the "not supported error" will be raised by the compressor wrapper when trying to write/read the data (see [here](https://github.com/joblib/joblib/blob/master/joblib/compressor.py#L226)).

Reply:
Thanks for the clarification.
All good to me then.